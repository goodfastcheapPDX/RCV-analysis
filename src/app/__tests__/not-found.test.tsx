import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import NotFound from "../not-found";

describe("Not Found Page", () => {
  it("should match snapshot", () => {
    const { container } = render(<NotFound />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
