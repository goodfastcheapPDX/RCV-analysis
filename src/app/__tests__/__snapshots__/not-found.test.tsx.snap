// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Not Found Page > should match snapshot 1`] = `
<div
  class="max-w-md mx-auto py-16"
>
  <div
    class="relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 [&>svg]:text-current bg-card text-card-foreground"
    data-slot="alert"
    role="alert"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-triangle-alert h-4 w-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
      />
      <path
        d="M12 9v4"
      />
      <path
        d="M12 17h.01"
      />
    </svg>
    <div
      class="col-start-2 line-clamp-1 min-h-4 font-medium tracking-tight"
      data-slot="alert-title"
    >
      Page Not Found
    </div>
    <div
      class="-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed mt-2"
      data-slot="alert-description"
    >
      The election or contest you're looking for doesn't exist or may have been moved.
    </div>
  </div>
  <div
    class="mt-6 flex gap-3 justify-center"
  >
    <a
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
      data-slot="button"
      href="/e"
    >
      Browse Elections
    </a>
    <a
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3"
      data-slot="button"
      href="/"
    >
      Go Home
    </a>
  </div>
</div>
`;
